File: Retos Abril/Java/Reto_13.java
package RetosAbril;

import java.util.Scanner;

import static java.lang.String.valueOf;

public class Reto13 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int hora = 0, minutos = 0, segundos = 0;

        try {
            System.out.print("Introduce el numero de segundos que deseas contar: ");
            int tiempo = Integer.parseInt(scanner.nextLine());

            // Construimos el mensaje con formato HH:MM:SS
            for (int i = 0; i < tiempo + 1; i++) {
                String mensaje = String.format("%02d:%02d:%02d", hora, minutos, segundos);

                // Evitamos imprimir cuando se llegue a los 60 segundos exactos (por lógica interna)
                if (i != 60) {
                    System.out.println(mensaje);
                }

                // Actualizamos el tiempo (cada 3600s = 1h, cada 60s = 1min)
                if (i % 3600 == 0 && i != 0){
                    hora++;
                    minutos = 0;
                    segundos = 0;
                } else if (i % 60 == 0 && i != 0) {
                    segundos = 0;
                    minutos++;
                }else{
                    segundos++;
                }
            }
        }catch (NumberFormatException e){
            System.out.println("Debes introducir un numero.");
        }
    }
}

File: Retos Abril/Python/Reto__13.py
import time

hora = 0
minuto = 0
segundos = 0

try:
    tiempo = int(input("Introduce el numero de segundos que desea contar: "))
    for i in range(0, tiempo + 1):
        time.sleep(1)  # Pausa el programa durante 1 segundo para simular el cronómetro

        # Imprime el tiempo en formato hh:mm:ss
        print(f"{"0" + str(hora) if hora < 10 else hora}:"  
              f"{"0" + str(minuto) if minuto < 10 else minuto}:"  
              f"{"0" + str(segundos) if segundos < 10 else segundos}")

        # Sí han pasado exactamente 3600 segundos (1 hora)
        if i % 3600 == 0 and i != 0:
            hora += 1  # Aumentamos la hora
            minuto = 0  # Reiniciamos minutos
            segundos = 0  # Reiniciamos segundos

        # Sí han pasado exactamente 60 segundos (1 minuto)
        elif i % 60 == 0 and i != 0:
            segundos = 0  # Reiniciamos segundos
            minuto += 1  # Aumentamos el minuto

        # En cualquier otro caso, simplemente incrementamos los segundos
        else:
            segundos += 1

except ValueError:
    print("Debes introducir un número.")




